atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1481,6 +1481,16 @@ public void createTables(final Map<TableReference, byte[]> tablesToMetadata) {
                 .collect(Collectors.toSet());
     }
 
+    @Override
+    public Iterable<TableReference> getLimitedTableNames(int maxResults) {
+        // Note: This is not optimal since getTableReferencesWithoutFiltering() loads *all* names, even though it
+        // returns a Stream. Getting a partial list of table names might not be possible with cassandra thrift.
+        return cassandraTables
+                .getTableReferencesWithoutFiltering()
+                .filter(tr -> !HiddenTables.isHidden(tr))
```

Comment:
Similar to above, should these lines (same as in `getAllTableNames()`)  be pulled out to a separate method that returns `Stream<TableReference>`?